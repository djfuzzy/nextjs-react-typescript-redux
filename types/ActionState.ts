export enum ActionState {
  Initial = 'Initial',
  Processing = 'Processing',
  Completed = 'Completed',
  Failed = 'Failed',
}
