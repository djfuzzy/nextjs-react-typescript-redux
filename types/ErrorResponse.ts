export type ErrorResponse = {
  title: string;
  content: string;
};

export const isErrorResponse = (
  error: ErrorResponse | string | undefined
): error is ErrorResponse => (error as ErrorResponse)?.title !== undefined;
