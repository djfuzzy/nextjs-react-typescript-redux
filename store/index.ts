import { combineReducers } from 'redux';
import { configureStore } from '@reduxjs/toolkit';

import commentReducer from './slices/comments';
import { ErrorResponse } from '../types/ErrorResponse';

const rootReducer = combineReducers({
  comments: commentReducer,
});

export type ErrorResult = {
  error: ErrorResponse | string;
};

export type ThunkConfig = { rejectValue: ErrorResult };

export type CoreState = ReturnType<typeof rootReducer>;

export default configureStore({
  reducer: rootReducer,
});
