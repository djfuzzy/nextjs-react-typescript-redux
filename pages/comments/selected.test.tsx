import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import SelectedCommentPage from './selected';
import { CoreState } from '../../store';
import { ActionState } from '../../types/ActionState';
import { Comment } from '../../types/Comment';

describe('Selected page', () => {
  const mockStore = configureStore<CoreState>();
  // TODO: Move this elsewhere and add test data
  const comments = {
    '1': {
      id: '1',
      name: 'Homer Simpson',
      content: 'Purple is a fruit.',
      createdAt: new Date('2021-03-22T14:52:10Z'),
    } as Comment,
  };

  const store = mockStore({
    comments: {
      entities: comments,
      ids: Object.keys(comments),
      loadState: ActionState.Completed,
      saveState: ActionState.Initial,
      selectedCommentId: '1',
    },
  });

  it('should match the snapshot', () => {
    const page = (
      <Provider store={store}>
        <SelectedCommentPage />
      </Provider>
    );
    const tree = renderer.create(page).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
