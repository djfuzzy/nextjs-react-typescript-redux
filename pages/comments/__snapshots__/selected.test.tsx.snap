// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selected page should match the snapshot 1`] = `
<section
  className="section"
>
  <div
    className="container"
  >
    <div
      className="comment-form"
    >
      <div
        className="card my-3"
      >
        <header
          className="card-header"
        >
          <p
            className="card-header-title"
          >
            Comment
          </p>
        </header>
        <div
          className="card-content"
        >
          <div
            className="content"
          >
            <form
              onSubmit={[Function]}
            >
              <div
                className="field"
              >
                <p>
                  Added 
                  03/22/2021
                   by
                   
                  <strong>
                    Homer Simpson
                  </strong>
                </p>
              </div>
              <div
                className="field"
              >
                <div
                  className="control"
                >
                  <textarea
                    className="textarea "
                    disabled={false}
                    maxLength={500}
                    name="content"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="e.g. These pretzils are making me thirsty."
                  />
                </div>
              </div>
              <div
                className="field is-grouped"
              >
                <div
                  className="control"
                >
                  <button
                    className="button is-link "
                    disabled={false}
                    type="submit"
                  >
                    Update
                  </button>
                </div>
                <div
                  className="control"
                >
                  <button
                    className="button is-link is-light"
                    onClick={[Function]}
                    type="button"
                  >
                    Cancel
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
